### PR TITLE
Adding OSGI tests for editor

### PR DESCRIPTION
**Pull request by minudika**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java [hunk 1]
```diff
@@ -68,12 +68,25 @@ private Option copySiddhiAppFileOption() {
...
 
+    private Option copySiddhiAppFileOption() {
+        Path carbonYmlFilePath;
+        String basedir = System.getProperty("basedir");
+        if (basedir == null) {
+            basedir = Paths.get(".").toString();
+        }
+        carbonYmlFilePath = Paths.get(basedir, "src", "test", "resources",
+                "editor", "siddhi-apps", "TestSiddhiApp.siddhi");
+        return copyFile(carbonYmlFilePath, Paths.get("wso2", "editor", "deployment", "workspace", "TestSiddhiApp.siddhi"));
```
Comment (tishan89) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java]: exceeding line length

File: components/osgi-tests/src/test/resources/editor/siddhi-apps/TestSiddhiApp.siddhi [hunk 2]
```diff
@@ -0,0 +1,8 @@
+@App:name('TestSiddhiApp')
+define stream FooStream (symbol string, price float, volume long);
+
+@source(type='inMemory', topic='symbol', @map(type='passThrough'))Define stream BarStream (symbol string, price float, volume long);
```
Comment (tishan89) [inline components/osgi-tests/src/test/resources/editor/siddhi-apps/TestSiddhiApp.siddhi]: Shall we put define statement to next line for better readability

File: components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java [hunk 3]
```diff
@@ -266,7 +319,7 @@ public void testCheckingWhetherInvalidFileExistsInWorkspace() throws Exception {
     }
 
     @Test
-    public void testFailureInExportingAFile() throws Exception {
+    public void testExportingAFile() throws Exception {
```
Comment (tishan89) [inline components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java]: Why are we changing a negative testcase to positive. Aren't we testing the negative behavior?

Comment (minudika) [reply components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java]: Negative scenario has been already tested in [this](https://github.com/minudika/carbon-analytics/blob/b6c1df854a4577a74c82aa4e7150bc61c93f3506/components/osgi-tests/src/test/java/org/wso2/carbon/analytics/test/osgi/SiddhiEditorTestCase.java#L352) method.

**Pull request merged**